File: Typescript3/1.R
fn remove_element(nums: &mut Vec<i32>, target: i32) -> usize {
    let mut pointer = 0;

    for num in nums.iter() {
        if *num != target {
            nums[pointer] = *num;
            pointer += 1;
        }
    }

    pointer
}

fn main() {
    let mut nums = vec![0, 1, 2, 2, 3, 0, 4, 2];
    let target = 2;
    let remove = remove_element(&mut nums, target);
    println!("Array after removing: {:?}", &nums[..remove]);
    println!("{}", remove);
}

File: Typescript3/1.py
def ofremove(a,target):

    pointer = 0
    
    for num in a:

        if  num != target:

            a[pointer] = num

            pointer +=1

    return pointer

nums = [0, 1, 2, 2, 3, 0, 4, 2]
target = 2
remove = ofremove(nums, target)
print("Array after removing:", nums[:remove])
print(remove)